fe1-web/src/features/rollCall/components/RollCallOpen.tsx

```diff
@@ -0,0 +1,175 @@
+  const generateToken = RollCallHooks.useGenerateToken();
+  const hasSeed = RollCallHooks.useHasSeed();
+  const toast = useToast();
+  const navigation = useNavigation<NavigationProps['navigation']>();
+
+  const [popToken, setPopToken] = useState('');
+  const [hasWalletBeenInitialized, setHasWalletBeenInitialized] = useState(hasSeed());
+
+  const onAddAttendees = useCallback(() => {
+    // Once the roll call is opened the first time, idAlias is defined
+    if (rollCall.idAlias) {
+      navigation.navigate(STRINGS.navigation_app_lao, {
+        screen: STRINGS.navigation_lao_events,
+        params: {
+          screen: STRINGS.navigation_lao_events_open_roll_call,
+          params: {
+            rollCallId: rollCall.id.toString(),
+            attendeePopTokens: (scannedPopTokens || []).map((e) => e.valueOf()),
+          },
+        },
+      });
+    } else {
+      toast.show(STRINGS.roll_call_location_error_scanning_no_alias, {
+        type: 'danger',
+        placement: 'bottom',
+        duration: FOUR_SECONDS,
+      });
+      console.debug(STRINGS.roll_call_location_error_scanning_no_alias);
+    }
+  }, [toast, navigation, rollCall, scannedPopTokens]);
+
+  const onCloseRollCall = useCallback(async () => {
+    // get the public key as strings from the existing rollcall
+    const previousAttendees = (rollCall.attendees || []).map((key) => key.valueOf());
+    // add the create a set of all attendees (takes care of deduplication)
+    const allAttendees = new Set([...previousAttendees, ...(scannedPopTokens || [])]);
+    // create PublicKey instances from the set of strings
+    const attendeesList = [...allAttendees].map((key: string) => new PublicKey(key));
+
+    if (!rollCall.idAlias) {
+      throw new Error('Trying to close a roll call that has no idAlias defined');
+    }
+
+    try {
+      await requestCloseRollCall(laoId, rollCall.idAlias, attendeesList);
+      navigation.navigate(STRINGS.navigation_lao_events_home);
+    } catch (err) {
+      console.log(err);
+      toast.show(STRINGS.roll_call_location_error_close_roll_call, {
+        type: 'danger',
+        placement: 'top',
+        duration: FOUR_SECONDS,
+      });
+    }
+  }, [toast, navigation, rollCall, laoId, scannedPopTokens]);
+
+  const toolbarItems: ToolbarItem[] = useMemo(() => {
+    if (!isOrganizer) {
+      return [];
+    }
+
+    return [
+      { title: STRINGS.roll_call_close, onPress: onCloseRollCall, buttonStyle: 'secondary' },
+      { title: STRINGS.roll_call_scan_attendees, onPress: onAddAttendees },
+    ] as ToolbarItem[];
+  }, [isOrganizer, onCloseRollCall, onAddAttendees]);
+
+  // re-check if wallet has been initialized after focus events
+  useEffect(() => {
+    // Return the function to unsubscribe from the event so it gets removed on unmount
+    return navigation.addListener('focus', () => {
+      setHasWalletBeenInitialized(hasSeed());
+    });
```

Comment:
This is quite subtle. An intermediate variable might make this easier to understand. 
```suggestion
   let unsubscribe = navigation.addListener('focus', () => {
      setHasWalletBeenInitialized(hasSeed());
    });
    
    // Return the function to unsubscribe from the event so it gets removed on unmount
    return unsubscribe;
```